From fc06cfd00b80e54fb178ffae51050675f42501b9 Mon Sep 17 00:00:00 2001
From: Inaky Perez-Gonzalez <inaky.perez-gonzalez@intel.com>
Date: Tue, 23 Nov 2010 15:55:37 -0800
Subject: [PATCH 07/20] supplicant: fix invocation of eap_peer_sm_init()

Fourth argument was being pass with wrong typing (and luckily it is
not really being used, so it wasn't pulling garbage). Fixed by
creating a null 'struct eap_config' that is passed to
eap_peer_sm_init().

Signed-off-by: Inaky Perez-Gonzalez <inaky.perez-gonzalez@intel.com>
---
 .../Agents/Supplicant/Source/SupplicantAgent.c     |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/InfraStack/OSAgnostic/WiMax/Agents/Supplicant/Source/SupplicantAgent.c b/InfraStack/OSAgnostic/WiMax/Agents/Supplicant/Source/SupplicantAgent.c
index 3948615..742c57b 100644
--- a/InfraStack/OSAgnostic/WiMax/Agents/Supplicant/Source/SupplicantAgent.c
+++ b/InfraStack/OSAgnostic/WiMax/Agents/Supplicant/Source/SupplicantAgent.c
@@ -466,6 +466,12 @@ wmx_Status_t InitSupplicantLibrary(VOID)
 	char answer[MAX_ANSWER_SIZE];
 	void *ft;
 #endif
+	struct eap_config eap_config_null = {
+		.opensc_engine_path = NULL,
+		.pkcs11_engine_path = NULL,
+		.pkcs11_module_path = NULL,
+		.wps = NULL,
+	};
 
 	TRACE(TR_MOD_SUPPLICANT_AGENT, TR_SEV_INFO,"Supplicant: InitSupplicantLibrary (IN)");
 	// init DS status variable
@@ -496,7 +502,7 @@ wmx_Status_t InitSupplicantLibrary(VOID)
 	eap_cb.notify_pending = peer_notify_pending;
 
 	eap_peer_register_methods();
-	eap_ctx.eap = eap_peer_sm_init(&eap_ctx, &eap_cb, &eap_ctx, &eap_ctx.eap_config);
+	eap_ctx.eap = eap_peer_sm_init(&eap_ctx, &eap_cb, &eap_ctx, &eap_config_null);
 	if (eap_ctx.eap == NULL)
 		return WMX_ST_FAIL;
 
-- 
1.7.4.1

